### PR TITLE
fix(leadspace): button sizing and layout for md leadspace

### LINKED ISSUE #9619: [Button group]: same width bug in lead space
**Engineering info:**  
- Innovation Team engineer: @m4olivei 
- Innovation Team JIRA Tickets: https://jsw.ibm.com/browse/ADCMS-2947
- DPO consulting engineer: N/A

-----

### Description

Lead space buttons should match width based on [Button group spec](https://ibm.ent.box.com/file/651713722198?s=xgqbas7e6gr14rvrropocrw75uooahm7): 
image

Current [storybook for Lead space](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-lead-space--super-with-image&knob-Button%201=Button%201&knob-Button%202=Button%202%20Button%202%20Quo%20usque%20tandem%20abutere%20&knob-Default%20image%20(defaultSrc):=1e7618ebe43eb5c8ce512709495c42fd.jpg&knob-Icon%201=ArrowRight20&knob-Icon%202=ArrowRight20&knob-Image%20alt%20text%20(alt):=Image%20alt%20text&knob-Link%201=https://example.com&knob-Link%202=https://example.com&knob-Number%20of%20buttons=2&knob-copy%20(copy):=Use%20this%20area%20for%20a%20short%20line%20of%20copy%20to%20support%20the%20title&knob-navigation%20elements%20(optional)=none&knob-title%20(title):=Heading%20can%20go%20on%20two%20lines%20max): 

image

### Component(s) impacted

Lead space

### Browser

Chrome

### Carbon for IBM.com version

1.24.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for Salesforce

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

NA

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by m4olivei**
### Related Ticket(s)

Resolves #9619 
Jira ticket: https://jsw.ibm.com/browse/ADCMS-2947

### Description

Adjustments made so that the width of buttons in a leadspace button group follows the largest one.

To test:

- [ ] Open up the Lead space > Super with Image story
- [ ] Adjust one or more button text such that the intrinsic button sizes would be different
- [ ] The width of all buttons should match the width of the button with the most content
- [ ] View at various breakpoints to ensure there are no regressions
- [ ] Open up the Button group > Default component
- [ ] View at various breakpoints to ensure there are no regressions

### Changelog

**Changed**

- Added a wrapper div around leadspace actions
- Use `display: grid` across all breakpoints for the leadspace button group

### CODE REVIEW

File: packages/web-components/src/components/button-group/button-group.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ class DDSButtonGroup extends StableSelectorMixin(LitElement) {
   }
 
   /**
-   * The CSS custom property name for the live button group item cout.
+   * The CSS custom property name for the live button group item count.
```
Comment (jkaeser) [inline packages/web-components/src/components/button-group/button-group.ts:76]: 👏

**Review by jkaeser — APPROVED** [review]
Nice work, @m4olivei! LGTM 👍

**Review by proeung — APPROVED** [review]
LGTM as well!

Screen Shot 2023-02-10 at 11 15 05 AM

Also, adding @oliviaflory and @shixiedesign (creator of the ticket) to do design QA for this PR.

**Comment by oliviaflory** [comment]
@m4olivei The desktop versions look great! 

Small fix request: At the small breakpoint the buttons are intended to span the full width of the browser, see the Percy differences below

Expecting (left) vs this PR (right)
Screenshot 2023-02-16 at 11 37 03 AM

File: packages/styles/scss/components/leadspace/_leadspace.scss [hunk 2]
```diff
@@ -153,6 +157,21 @@ $btn-min-width: 26;
...
 
+    .#{$prefix}--leadspace__action {
+      display: inline-block;
+
+      // Button requirements dictate buttons should be full width at 320px and
+      // down. The carbon--breakpoint-down(sm) mixin has logic that will use
+      // a value of 320px - 0.02, which is not quite what we want, so instead,
+      // we use carbon--breakpoint-down with a number argument, using the
+      // defined width for the sm breakpoint.
+      $breakpoint: map-get($carbon--grid-breakpoints, sm);
```
Comment (m4olivei) [inline packages/styles/scss/components/leadspace/_leadspace.scss:168]: @proeung @jkaeser Is this an OK approach to you?

Comment (jkaeser) [reply packages/styles/scss/components/leadspace/_leadspace.scss:168]: I'm actually not sure!

I imagine using `carbon--breakpoint-down(sm)` causes this style to apply up to 319px, not 320px, and I wonder if that's actually okay. Assuming the `@carbon/grid` package is compliant with the design spec, perhaps this is the designed behavior and it's unwise to deviate from it.

On the other hand, I trust the designers understand the intricacies of the design system way better than me, and their specs (below) should be followed to the letter.

Screen Shot 2023-02-20 at 5 00 12 PM

Comment (m4olivei) [reply packages/styles/scss/components/leadspace/_leadspace.scss:168]: > I imagine using carbon--breakpoint-down(sm) causes this style to apply up to 319px, not 320px

Yes, that's the effect in practice when I tried it locally.

> On the other hand, I trust the designers understand the intricacies of the design system way better than me, and their specs (below) should be followed to the letter.

I agree, in which case, this was the only option I could come up with. It feels safe to me, we would inherit any changes to the default set of breakpoints in the same way we would if we were able to use the mixins directly.

**Comment by m4olivei** [comment]
Hi @oliviaflory :wave: . Thanks for the feedback! I've addressed it, and Percy is no longer flagging a difference in the button widths. Please have another look when you have a chance. Thanks.

**Comment by oliviaflory** [comment]
Thanks @m4olivei ! Last question, any ideas why the [Web Components HTML Test App](https://percy.io/538fc19a/Carbon-for-IBM.com-Web-Components-HTML-Test-App-Upstream/builds/25232361/changed/1408050680?browser=chrome&browser_ids=34&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280) has really wide buttons now? Almost looks like the mobile styles are being applied to those templates too?

**Comment by m4olivei** [comment]
> Thanks @m4olivei ! Last question, any ideas why the [Web Components HTML Test App](https://percy.io/538fc19a/Carbon-for-IBM.com-Web-Components-HTML-Test-App-Upstream/builds/25232361/changed/1408050680?browser=chrome&browser_ids=34&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280) has really wide buttons now? Almost looks like the mobile styles are being applied to those templates too?

Looks like the leadspace component is not being used in the expected way in this HTML Test Application. The button-group child element is not being assigned to the expected slot, so it doesn't pick up the relevant styles. I'm not familiar with how that HTML Test Application is being built. @proeung could you help me identify where that is? If that's not sufficient, eg. to fix the HTML Test Application, I'd have to go back and see how to apply the styles more generally, which should be a big deal, just want to understand that HTML Test Application better first.

**Comment by proeung** [comment]
> Looks like the leadspace component is not being used in the expected way in this HTML Test Application. The button-group child element is not being assigned to the expected slot, so it doesn't pick up the relevant styles. I'm not familiar with how that HTML Test Application is being built. @proeung could you help me identify where that is?

@m4olivei The Web Components HTML Test Application (see [repo](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-web-components-test)) is managed in a different repo and it's used to identify visual regression as part of the CI checks. Also, you're correct that the markup in the test app is missing the expected slot (see [code](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-web-components-test/blob/main/src/pages/example-page-a/example-page-a.hbs#L6) and [example](https://carbon-design-system.github.io/carbon-for-ibm-dotcom-web-components-test/canary/example-page-a/index.html)).

@oliviaflory @annawen1 Does your team has a process for updating the HTML Test App to reflect changes coming from this mono repo? For instance, do we need to address this in the test app first before the changes in the PR can be merged down?

**Comment by m4olivei** [comment]
Thanks @proeung !

Here is a PR for the Web Component Test example repo that changes the `<dds-leadspace>` component usage to the correct use of the action slot: https://github.com/carbon-design-system/carbon-for-ibm-dotcom-web-components-test/pull/131

**Review by annawen1 — APPROVED** [review]
LGTM! Thanks for the detailed testing steps 👍

**Comment by annawen1** [comment]
@oliviaflory are we good to merge this PR in?

**Comment by oliviaflory** [comment]
@annawen1 looks good to me! Thanks @m4olivei and @proeung for clarifying!

**Pull request merged**